pangeo_forge_recipes/transforms.py

```diff
@@ -82,12 +103,51 @@ def _drop_keys(kvp):
 
 @dataclass
 class DropKeys(beam.PTransform):
-    """Simple transform to remove keys for use in a Kerchunk Reference Recipe Pipeline"""
+    """Simple transform to remove keys."""
 
     def expand(self, pcoll):
         return pcoll | "Drop Keys" >> beam.Map(_drop_keys)
 
 
+def _assign_concurrency_group(elem, max_concurrency: int):
+    return (random.randint(0, max_concurrency - 1), elem)
+
+
+@dataclass
+class MapWithConcurrencyLimit(beam.PTransform):
+    """A transform which maps calls to the provided function, optionally limiting the maximum
+    number of concurrent calls. Useful for situations where the provided function requests data
+    from an external service that does not support an unlimited number of concurrent requests.
+
+    :param map_func: The function to pass to beam.Map (in the case of no concurrency limit)
+      or beam.FlatMap (if max_concurrency is specified).
+    :param op_name: The name of the operation performed by `map_func`.
+    :param map_func_kws: Kwargs passed to all invocations of `map_func`.
+    :param max_concurrency: The maximum number of concurrent invocations of `map_func`.
+      If unspecified, no limit is imposed by this transform (therefore the concurrency
+      limit will be set by the Beam Runner's configuration).
```

Comment:
I would try to make the signature of this PTransform stay as close as possible to that of the built-in Map transform: https://beam.apache.org/releases/pydoc/2.49.0/apache_beam.transforms.core.html#apache_beam.transforms.core.Map

Reply:
✅ https://github.com/pangeo-forge/pangeo-forge-recipes/pull/557/commits/95a9af2caad15904722bcacc22999c0014d5e192

Good call. This looks better.